.github/workflows/malware-scan.yml

```diff
@@ -0,0 +1,53 @@
+# (C) Copyright 2023 Hewlett Packard Enterprise Development LP
+
+name: Malware Scan
+on:
+  pull_request:
+  push:
+    branches:
+      - main
+      - master
+
+jobs:
+  scan_repository:
+    name: "Scan Virus on Repository"
+    runs-on: "ubuntu-latest"
+    permissions:
+      id-token: write
+      contents: read
+    steps:
+      - name: "Checkout Repository"
+        uses: "actions/checkout@v3.5.0"
+        with:
+          fetch-depth: 0
+      - name: "Scan Source for Viruses"
+        id: scan
+        run: |
+          SHA_SHORT=$(git rev-parse --short HEAD)
+          SCAN_OUTPUT=/tmp/clamav-repository-results-${{ github.event.repository.name }}-${SHA_SHORT}-$(date +%s).txt
+          echo "scan_output=${SCAN_OUTPUT}" >> $GITHUB_OUTPUT
+          docker run --rm -v ${{ github.workspace }}:/scandir -v /tmp:/tmp -e SCAN_OUTPUT=${SCAN_OUTPUT} clamav/clamav:stable clamscan --infected --recursive --max-files=0 --max-filesize=0 --max-scansize=0 --max-recursion=1000 --max-dir-recursion=1000 /scandir | tee -a $SCAN_OUTPUT
+          INFECTED_FILES=$(grep ^Infected $SCAN_OUTPUT | awk '{print $NF}')
+          echo "infected_files=${INFECTED_FILES}" >> $GITHUB_OUTPUT
+          exit $INFECTED_FILES
+      - name: "Configure AWS Credentials"
+        if: always()
+        uses: aws-actions/configure-aws-credentials@v2
+        with:
+          role-to-assume: arn:aws:iam::522637239241:role/s3-avscan-upload 
```

Comment:
are these HPE AWS roles ? if yes, are we fine to put them in public repos.

Reply:
yes its HPE's AWS roles, I asked the same to the security guy who suggested. But not reply back. So, if they come back saying we shouldn't then we can keep in secrets. Or what do you suggest?

Reply:
These repos are open to public and can be accessed by anyone. I do not recommend to expose any users to public repos. We are ideally exposing our AWS Account ID and then the roles. @reubenur-rahman  do you have any suggestions here ?

Reply:
from Security team : This should be ok because our OIDC role and policy only allows to post the results to the bucket and even though public, a peer reviewed pull request would still be required; being public is all the mor reason to run the scan; I have approved the PR.